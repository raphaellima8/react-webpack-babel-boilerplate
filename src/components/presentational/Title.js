import React from 'react';
import styled from 'styled-components';

const HeaderTitle = styled.h1`
  margin: 0;
  font-size: 1rem;

  @media (min-width: 600px) {
    font-size: 2rem;
  }
`;

const Title = ({ text }) => <HeaderTitle>{text}</HeaderTitle>;

export default Title;
