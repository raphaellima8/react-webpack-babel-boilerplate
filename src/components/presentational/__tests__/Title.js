import React from 'react';
import 'jest-styled-components';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Title from '../Title';

describe('Title Component tests', () => {
  it('Should render component with received props', () => {
    const componentRef = shallow(<Title text="texto teste" />);
    expect(componentRef).toMatchSnapshot();
    expect(componentRef.find('Title__HeaderTitle').text()).toBe('texto teste');
  });
});

describe('Test styled component', () => {
  it('Match style rules snapshot', () => {
    const componentRef = renderer.create(<Title text="texto teste" />).toJSON();
    expect(componentRef).toMatchSnapshot();
    expect(componentRef).toHaveStyleRule('margin', '0');
    expect(componentRef).toHaveStyleRule('font-size', '1rem');
    expect(componentRef).toHaveStyleRule('font-size', '2rem', {
      media: '(min-width: 600px)'
    });
  });
});
