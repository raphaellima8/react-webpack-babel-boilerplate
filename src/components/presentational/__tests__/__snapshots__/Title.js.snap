// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test styled component Match style rules snapshot 1`] = `
.c0 {
  margin: 0;
  font-size: 1rem;
}

@media (min-width:600px) {
  .c0 {
    font-size: 2rem;
  }
}

<h1
  className="c0"
>
  texto teste
</h1>
`;

exports[`Title Component tests Should render component with received props 1`] = `
<Title__HeaderTitle>
  texto teste
</Title__HeaderTitle>
`;
